packages/playwright-test/src/reporters/base.ts

```diff
@@ -371,10 +371,28 @@ export function formatTestTitle(config: FullConfig, test: TestCase, step?: TestS
   return `${projectTitle}${location} › ${titles.join(' › ')}${stepSuffix(step)}`;
 }
 
-function formatTestHeader(config: FullConfig, test: TestCase, indent: string, index?: number): string {
+function formatTestHeader(config: FullConfig, test: TestCase, options: { indent?: string, index?: number, mode?: 'default' | 'error' } = {}): string {
   const title = formatTestTitle(config, test);
-  const header = `${indent}${index ? index + ') ' : ''}${title}`;
-  return separator(header);
+  const header = `${options.indent || ''}${options.index ? options.index + ') ' : ''}${title}`;
+  let fullHeader = header;
+
+  // Render the path to the deepest failing test.step.
+  if (options.mode === 'error' && test.results.length === 1 && test.results[0].errors.length) {
+    const stepPath: string[] = [];
+    const visit = (steps: TestStep[]) => {
+      const errors = steps.filter(s => s.error);
+      if (errors.length > 1)
+        return;
+      if (errors.length === 1 && errors[0].category === 'test.step') {
+        stepPath.push(errors[0].title);
+        visit(errors[0].steps);
+      }
+    };
+    visit(test.results[0].steps);
+    fullHeader += ['', ...stepPath].join(' › ');
```

Comment:
We have `stepSuffix` helper in this file already.

Reply:
Unfortunately it does not help much :)